tools/linker/CoreOptimizeGeneratedCode.cs

```diff
@@ -73,27 +73,30 @@ public override bool IsActiveFor (AssemblyDefinition assembly)
 			}
 			
 			// if the assembly does not refer to [CompilerGeneratedAttribute] then there's not much we can do
-			HasGeneratedCode = false;
+			HasOptimizableCode = false;
 			foreach (TypeReference tr in assembly.MainModule.GetTypeReferences ()) {
 				if (tr.Is ("System.Runtime.CompilerServices", "CompilerGeneratedAttribute")) {
 #if DEBUG
 					Console.WriteLine ("Assembly {0} : processing", assembly);
 #endif
-					HasGeneratedCode = true;
+					HasOptimizableCode = true;
+					break;
```

Comment:
I think we can _eventually_ (existing bindings) kill the above condition since we now have a better, more accurate, way to identify what can be optimized.
Might be worth putting it under `XAMCORE_4_0` so we don't miss the opportunity

Reply:
Fixed: 53ea968d0382cb45aeadd271575d6c212b3a5825 (I had to some extra XAMCORE_4_0 logic to mtouch/mmp, since they're a single binary and a conditional compilation directive wouldn't work as expected).